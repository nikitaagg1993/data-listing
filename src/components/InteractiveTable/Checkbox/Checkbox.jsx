/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import PropTypes from 'prop-types';

const Checkbox = ({
  checked,
  name,
  onClick,
}) => (
  <input
    type="checkbox"
    onClick={onClick}
    name={name}
    aria-label={name}
    checked={checked}
    onChange={() => null}
  />
);

Checkbox.propTypes = {
  name: PropTypes.string.isRequired,
  checked: PropTypes.bool,
  onClick: PropTypes.func,
};

Checkbox.defaultProps = {
  checked: false,
  onClick: null,
};

export default Checkbox;
