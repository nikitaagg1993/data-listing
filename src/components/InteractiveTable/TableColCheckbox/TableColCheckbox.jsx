import React, { useCallback } from 'react';
import PropTypes from 'prop-types';
import { useTableContext } from '../DataTableContext';

import Checkbox from '../Checkbox';
import TableColStyle from './styled';

const TableColCheckbox = ({ head }) => {
  const {
    dispatch,
    data,
    allSelected,
    keyField,
    mergeSelections,
  } = useTableContext();

  // The row count should subtrtact rows that are disabled
  const rowCount = data.length;

  const handleSelectAll = useCallback(() => dispatch({
    type: 'SELECT_ALL_ROWS',
    rows: data,
    rowCount,
    mergeSelections,
    keyField,
  }), [dispatch, keyField, mergeSelections, rowCount, data]);

  return (
    <TableColStyle
      head={head}
      noPadding
    >
      <Checkbox
        name="select-all-rows"
        onClick={handleSelectAll}
        checked={allSelected}
      />
    </TableColStyle>
  );
};

TableColCheckbox.propTypes = {
  head: PropTypes.bool,
};

TableColCheckbox.defaultProps = {
  head: true,
};
export default TableColCheckbox;
