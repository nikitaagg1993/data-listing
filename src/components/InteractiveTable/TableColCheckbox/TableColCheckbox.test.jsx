import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import TableColCheckbox from './TableColCheckbox';
import Checkbox from '../Checkbox';

jest.mock('../DataTableContext', () => ({
  useTableContext: () => ({
    dispatch: jest.fn(),
    data: [],
  }),
}));

test('Table Col Checkbox', () => {
  const tree = renderer.create(<TableColCheckbox head />);
  const testInstance = tree.root;

  expect(tree.toJSON()).toMatchSnapshot();
  expect(testInstance.findAllByType(Checkbox).length).toEqual(1);
  expect(testInstance.props).toMatchObject({ head: true });
});
