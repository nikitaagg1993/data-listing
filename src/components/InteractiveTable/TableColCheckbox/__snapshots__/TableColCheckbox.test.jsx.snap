// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Col Checkbox 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding: 0;
}

.c1 {
  -webkit-flex: 0 0 48px;
  -ms-flex: 0 0 48px;
  flex: 0 0 48px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
}

<div
  className="c0 c1"
>
  <input
    aria-label="select-all-rows"
    checked={false}
    name="select-all-rows"
    onChange={[Function]}
    onClick={[Function]}
    type="checkbox"
  />
</div>
`;
